src/plugins/intel_gna/src/transformations/gather_sinking_utils.cpp

```diff
@@ -0,0 +1,285 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/gather_sinking_utils.hpp"
+
+#include <openvino/pass/pattern/op/or.hpp>
+#include <transformations/utils/utils.hpp>
+#include <utility>
+
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/label.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/util/common_util.hpp"
+#include "openvino/util/log.hpp"
+#include "transformations/rt_info/gather_sinking_attr.hpp"
+#include "transformations/rt_info/gather_sinking_attr.hpp"
+
+namespace gather_sinking {
+
+using namespace ov;
+using namespace ov::opset9;
+
+using NodePtr = std::shared_ptr<Node>;
+
+GatherInputsInfo GetFirstGatherInput(NodePtr node) {
+    for (size_t input_idx = 0; input_idx < node->get_input_size(); ++input_idx) {
+        NodePtr input_node = node->get_input_node_shared_ptr(input_idx);
+        auto gather_node = as_type_ptr<Gather>(input_node);
+        if (!gather_node)
+            continue;
+        auto indices_const_node = as_type_ptr<Constant>(gather_node->input_value(1).get_node_shared_ptr());
+        if (!indices_const_node)
+            continue;
+        auto axes_const_node = as_type_ptr<Constant>(gather_node->input_value(2).get_node_shared_ptr());
+        if (!axes_const_node)
+            continue;
+        {
+            GatherInputsInfo input_info;
+            input_info.gather = gather_node;
+            input_info.indices_const = indices_const_node;
+            input_info.axes_const = axes_const_node;
+            input_info.input_idx = input_idx;
+            return input_info;
+        }
+    }
+
+    return GatherInputsInfo();
+}
+
+bool IfNodeHasGatherInputs(const Output<Node>& output) {
+    GatherInputsInfo inputs_info = GetFirstGatherInput(output.get_node_shared_ptr());
+    return !inputs_info.isEmpty();
+}
+
+namespace {
+
+bool HasDynamicRankInput(NodePtr node) {
+    for (auto& input_node : node->input_values()) {
+        const ov::Rank output_rank = input_node.get_partial_shape().rank();
+        if (output_rank.is_dynamic())
+            return true;
+    }
+    return false;
+}
+
+ov::Rank::value_type GetMaxInputRank(const NodePtr& node) {
+    ov::Rank::value_type max_input_rank = 0;
+    for (auto& input_node : node->input_values()) {
+        const ov::Rank output_rank = input_node.get_partial_shape().rank();
+        if (output_rank.is_dynamic())
+            return -1;
+        const ov::Rank::value_type output_rank_len = output_rank.get_length();
+        if (output_rank_len > max_input_rank)
+            max_input_rank = output_rank_len;
+    }
+    return max_input_rank;
+}
+
+NodePtr InsertUnsqueeze(Output<Node> node, size_t n_dims) {
```

Comment:
```suggestion
NodePtr InsertUnsqueeze(ov::Output<ov::Node> node, size_t n_dims) {
```

Reply:
Could you please align names in the code (ov::Node everywhere or Node but everywhere).
And not clear for me we can't pass Node argument, it would allow us to avoid unnecessary conversions Node->Output<Node>

Reply:
never mind, I saw that we need to process ov::Output<ov::Node> in the common code.